### PR TITLE
[TechDocs] Sync entityRef and path updates

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Update the `path` at the same time the `entityRef` is updated to prevent the path update from happening before the entityRef is updated and we end up dispatching an unnecessary API request to the old entity name with the new path which results in 404 error.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs-react/src/context.tsx [hunk 1]
```diff
@@ -96,6 +102,7 @@ export type TechDocsReaderPageProviderRenderFunction = (
  * @public
  */
 export type TechDocsReaderPageProviderProps = {
+  path?: string;
```
Comment (emmaindal) [inline plugins/techdocs-react/src/context.tsx:105]: Maybe this needs to be required? And if it is, the changes is probably breaking?

File: plugins/techdocs-react/src/context.tsx [hunk 2]
```diff
@@ -144,7 +152,10 @@ export const TechDocsReaderPageProvider = memo(
     );
   },
   (prevProps, nextProps) => {
-    return areEntityRefsEqual(prevProps.entityRef, nextProps.entityRef);
+    return (
+      areEntityRefsEqual(prevProps.entityRef, nextProps.entityRef) &&
+      arePathsEqual(prevProps.path, nextProps.path)
```
Comment (emmaindal) [inline plugins/techdocs-react/src/context.tsx:157]: Maybe worth adding a test for this change too, to make sure the path is what we expect.

File: plugins/techdocs-react/src/context.tsx (hunk 1 above)
Comment (camilaibs) [reply plugins/techdocs-react/src/context.tsx:107]: I added fallback for path and entityId which are derived from URL, what do you think?

**Review by freben — APPROVED** [review]
approved to unblock!

**Comment by iamEAP** [comment]
Thanks @camilaibs!  I attempted to reproduce the type of error we were seeing internally in this repo, but had trouble doing so in isolation.  While this seems like a reasonable clean up, I'm hesitant to merge it in if it's not strictly necessary (especially given the small API addition, as Emma mentioned).

Whaddya say if I close this for now, but we can re-evaluate if this starts to more actively cause issues?

**Comment by camilaibs** [comment]
> Thanks @camilaibs! I attempted to reproduce the type of error we were seeing internally in this repo, but had trouble doing so in isolation. While this seems like a reasonable clean up, I'm hesitant to merge it in if it's not strictly necessary (especially given the small API addition, as Emma mentioned).
> 
> Whaddya say if I close this for now, but we can re-evaluate if this starts to more actively cause issues?

Closing 🙌🏻

**Pull request closed without merging**